src/main/java/org/dependencytrack/tasks/AbstractNistMirrorTask.java

```diff
@@ -55,6 +59,15 @@ Vulnerability synchronizeVulnerability(final QueryManager qm, final Vulnerabilit
                 final Map<String, PersistenceUtil.Diff> diffs = updateVulnerability(persistentVuln, vuln);
                 if (!diffs.isEmpty()) {
                     logger.debug("%s has changed: %s".formatted(vuln.getVulnId(), diffs));
+
+                    // Check if we need to emit a vulnerability changed event
+                    if (diffs.containsKey("severity") && diffs.get("severity") != null) {
+                        final PersistenceUtil.Diff severityDiff = diffs.get("severity");
+                        if (severityDiff.before() instanceof final Severity oldSeverity && severityDiff.after() instanceof final Severity newSeverity) {
+                            Event.dispatch(new ProjectVulnerabilityUpdateEvent(persistentVuln, new VulnerabilityUpdateDiff(oldSeverity, newSeverity)));
+                        }
+                    }
```

Comment:
Since this code executes within a transaction, it *can* happen that we emit an event, but committing the transaction fails, leaving us in an inconsistent state. Consider moving the event dispatch outside of the transaction block.